var/spack/repos/builtin/packages/py-build/package.py

```diff
@@ -16,7 +16,7 @@ class PyBuild(PythonPackage):
 
     variant('virtualenv', default=False, description='Install optional virtualenv dependency')
 
-    depends_on('python@3.6:', type=('build', 'run'))
+    depends_on('python@3.6:+ensurepip', type=('build', 'run'))
```

Comment:
How would we add a test case for this functionality? I could add it in a followup if it's too tedious here.

Reply:
You could create a fake package following the instructions at https://packaging.python.org/en/latest/tutorials/packaging-projects/ and try to build it. I think it would need at least one dependency? It would also need internet access.